base_layer/core/src/covenants/arguments.rs

```diff
@@ -215,6 +228,8 @@ impl CovenantArg {
 
     require_x_impl!(require_outputfields, OutputFields, "outputfields");
 
+    /// Require that the given `CovenantArg` instance is of the form
+    /// `CovenantArg::Bytes` and extracts the underlying value.
```

Comment:
Due to `$output_type:ty` we can now replace `pub fn require_bytes(self) -> Result<Vec<u8>, CovenantError> {` with
```suggestion
require_x_impl!(require_bytes, Bytes, "bytes", Vec<u8>);
```